eclipse-che/che-server
# feat: Create project with Che-server's SA on OpenShift with OAuth

### What does this PR do?
Create project with Che-server's SA on OpenShift with OAuth

### Screenshot/screencast of this PR
n/a

### What issues does this PR fix or reference?
Fixes https://github.com/eclipse/che/issues/19717

### How to test this PR?
1. Deploy new che-server image `quay.io/skabashn/che-server:che19717_v5`
2. Enable `CHE_INFRA_OPENSHIFT_PROJECT_INIT__WITH__SERVER__SA=true`
3. Disable self-provisioning 
```
oc patch clusterrolebinding.rbac self-provisioners -p '{"subjects": null}'
oc patch clusterrolebinding.rbac self-provisioners -p '{ "metadata": { "annotations": { "rbac.authorization.kubernetes.io/autoupdate": "false" } } }'

```
4. Login as a regular user
5. Start a new workspace.
6. OpenshiftProject should be created sucessfully.
### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
